baybe/searchspace/continuous.py

```diff
@@ -29,16 +31,18 @@ class SubspaceContinuous(SerialMixin):
     parameter views.
     """
 
-    parameters: List[NumericalContinuousParameter] = field(
-        validator=lambda _1, _2, x: validate_parameter_names(x)
+    parameters: Tuple[NumericalContinuousParameter, ...] = field(
```

Comment:
It's not a `mypy` thing at all but I think an important change to prevent users from running into problems that I can already foresee to happen. In essence, it builds the third pillar to making our building blocks fully immutable and thus (almost) impossible to break by a user:
1. With every `mypy` refactoring, I made the classes `frozen` to avoid that people directly modify attributes
2. We use slots everywhere to prevent people adding attributes by mistake (which happened to me already several types by misspelling the names of our old `allow...*` flags), which gives you the worst kind of silent bugs
3. This change makes the attributes themselves immutable.

The kind of error that I already foresee can happen everywhere, but one arbitrary example could be that people try to "expand" the parameter range on the fly, depending on their measured results. Something in the style of:
```python
p1 = NumericalDiscreteParameter("param", values=[1,2,3])
searchspace = SearchSpace.from_product([p1, ...])
# do some experimentation
p1.values.append(4)
```
This would yield hard-to-detect bugs because the code would not to what people might think it does. Therefore, preventing this from the getgo seems the right approach to me. After all, immutability is a really useful concept, and  tuples are simply to be preferred over lists if you don't intend to change them or want to forbid it (i.e. that's exactly the purpose of them).

You will notice that the same transition already happened in the other `mypy` refactorings as well and I wanted to roll this out for all basic building blocks.

What are your thoughts?

Reply:
ok these arguments make sense and I'd support hat.

It is always welcome to announce changes that are not directly related to the main PR intent in the description or via a preceding discussion, not only on request